pkg/resources/kafka/pod.go

```diff
@@ -75,9 +75,9 @@ func (r *Reconciler) pod(id int32, brokerConfig *v1beta1.BrokerConfig, pvcs []co
 
 	dataVolume, dataVolumeMount := generateDataVolumeAndVolumeMount(pvcs)
 
-	var volume []corev1.Volume
-	var volumeMount []corev1.VolumeMount
-	var initContainers []corev1.Container
+	volume := brokerConfig.Volumes
+	volumeMount := brokerConfig.VolumeMounts
+	initContainers := brokerConfig.InitContainers
```

Comment:
To be on the safe side it worth sorting the content of `volume`, `volumeMount` and `initContainers` to ensure that the broker pod is not restarted unnecessary between two reconciles. E.g. user changes the ordering of volumes in `brokerConfig.Volumes` that should cause an update to the pod. If object matcher can handle this already than no need for the sorting.